### PR TITLE
fix for windows

### PR DESCRIPTION
**Pull request by Andrey-Russe**
fix for windows build.
added manuals for environment setup: Windows, MacOS, Linux
Linux - TODO

### CODE REVIEW

File: stout/semaphore.h [hunk 1]
```diff
@@ -2,13 +2,14 @@
 
 #ifdef __MACH__
 #include <mach/mach.h>
-#elif __WINDOWS__
+#elif _WIN32
 #else
```
Comment (benh) [inline stout/semaphore.h:6]: `#include <windows.h> // For Windows semaphore.`

And update line 54 to be `_WIN32` not `_WINDOWS_`.

File: test/scheduler.cc [hunk 2]
```diff
@@ -1,3 +1,5 @@
+#ifndef _WIN32 //temp solution for WINDOWS.
```
Comment (benh) [inline test/scheduler.cc:1]: Hopefully don't need this now that we include correct file above.

File: stout/callback.h [hunk 3]
```diff
@@ -1,5 +1,9 @@
 #pragma once
 
+# if defined(_WIN32)
```
Comment (benh) [inline stout/callback.h:3]: Can we use `#ifdef` to be consistent with all the other ones?

File: BUILD.bazel [hunk 4]
```diff
@@ -35,8 +35,5 @@ cc_library(
     srcs = [
         "stout/scheduler.cc",
     ],
-    deps = [
```
Comment (benh) [inline BUILD.bazel:38]: This will break Linux and Mac OS builds now?

**Review by benh — CHANGES_REQUESTED** [review]

**Comment by benh** [comment]
Hey @Andrey-Russe, are you able to update this pull request with the changes requested?

File: stout/semaphore.h (hunk 1 above)
Comment (Andrey-Russe) [reply stout/semaphore.h:8]: It is done.

File: BUILD.bazel (hunk 4 above)
Comment (Andrey-Russe) [reply BUILD.bazel:38]: It is cancelled.

**Comment by benh** [comment]
This was merged after some tweaks in 5eb7153b88893eca2b9f53f6e3987dd65a2792c1.

**Pull request closed without merging**